python/pyspark/pandas/indexes/base.py

```diff
@@ -1992,6 +1992,16 @@ def difference(self, other: "Index", sort: Optional[bool] = None) -> "Index":
         """
         from pyspark.pandas.indexes.multi import MultiIndex
 
+        if isinstance(other, Index) and not isinstance(self, type(other)):
```

Comment:
can you extract this condition and define it in a variable? It's difficult to follow the if-elses here.